applications/PfemFluidDynamicsApplication/custom_elements/two_step_updated_lagrangian_V_P_element.cpp

```diff
@@ -814,6 +793,47 @@ void TwoStepUpdatedLagrangianVPElement<TDim>::CalculateDeltaPosition(Matrix & rD
     KRATOS_CATCH( "" )
 }
 
+  template<>
+  void TwoStepUpdatedLagrangianVPElement<2>::GetDisplacementValues(Vector& rValues,
+								   const int Step)
+  {
+    GeometryType& rGeom = this->GetGeometry();
+    const SizeType NumNodes = rGeom.PointsNumber();
+    const SizeType LocalSize = 2*NumNodes;
+
+    if (rValues.size() != LocalSize) rValues.resize(LocalSize);
+
+    SizeType Index = 0;
+
+    for (SizeType i = 0; i < NumNodes; ++i)
+      {
+        rValues[Index++] = rGeom[i].FastGetSolutionStepValue(DISPLACEMENT_X,Step);
+        rValues[Index++] = rGeom[i].FastGetSolutionStepValue(DISPLACEMENT_Y,Step);
+      }
```

Comment:
It is faster to take 'rGeom[i].FastGetSolutionStepValue(DISPLACEMENT,Step)' once, by reference, then extract each component.